builtin/game/item_entity.lua

```diff
@@ -189,16 +190,16 @@ core.register_entity(":__builtin:item", {
 		-- Push item out when stuck inside solid node
 		if is_stuck then
 			local shootdir
-			local cx = pos.x % 1
-			local cz = pos.z % 1
+			local cx = (pos.x % 1) - 0.5
+			local cz = (pos.z % 1) - 0.5
 			local order = {}
 
 			-- First prepare the order in which the 4 sides are to be checked.
 			-- 1st: closest
 			-- 2nd: other direction
 			-- 3rd and 4th: other axis
 			local cxcz = function(o, cw, one, zero)
-				if cw > 0 then
+				if cw < 0 then
 					table.insert(o, { [one]=1, y=0, [zero]=0 })
 					table.insert(o, { [one]=-1, y=0, [zero]=0 })
```

Comment:
Spacing needed around `=`. Should've pointed this out earlier, sorry. :)

Reply:
Also, no spaces just inside curly brackets.